spec/models/job_proxy_dispatcher_spec.rb

```diff
@@ -276,45 +271,139 @@
+        expect(dispatcher.busy_proxies).to eq "MiqServer_#{@server.id}" => 2
+        allow(dispatcher).to receive(:embedded_scan_resource).and_return(nil)
+        dispatcher.assign_proxy_to_job(@server, job)
+        expect(dispatcher.busy_proxies).to eq "MiqServer_#{@server.id}" => 3
+      end
 
-        allow(@proxy_dispatcher).to receive(:embedded_scan_resource).and_return(nil)
-        @proxy_dispatcher.assign_proxy_to_job(@server, job)
-        expect(@proxy_dispatcher.busy_proxies).to eq "MiqServer_#{@server.id}" => 3
+      it "links job to instance of MiqServer and updates :started_on and :dispatch_status atributes" do
+        allow(dispatcher).to receive(:embedded_scan_resource).and_return(nil)
+        Timecop.freeze do
+          timestamp = Time.now.utc
+          dispatcher.assign_proxy_to_job(@server, job)
+          expect(job.started_on).to eq timestamp
+        end
+        expect(job.miq_server_id).to eq @server.id
+        expect(job.dispatch_status).to eq "active"
       end
     end
   end
 
   describe "#start_job_on_proxy" do
     it "creates job options and passing it to `queue_signal'" do
       job = Job.create_job("VmScan", :agent_id => @server.id, :name => "Hello, World")
-      proxy_dispatcher = JobProxyDispatcher.new
-      proxy_dispatcher.instance_variable_set(:@active_vm_scans_by_zone, @server.my_zone => 0)
+      dispatcher.instance_variable_set(:@active_vm_scans_by_zone, @server.my_zone => 0)
 
       job_options = {:args => ["start"], :zone => @server.my_zone, :server_guid => @server.guid, :role => "smartproxy"}
-      expect(proxy_dispatcher).to receive(:assign_proxy_to_job)
-      expect(proxy_dispatcher).to receive(:queue_signal).with(job, job_options)
+      expect(dispatcher).to receive(:assign_proxy_to_job)
+      expect(dispatcher).to receive(:queue_signal).with(job, job_options)
+
+      dispatcher.start_job_on_proxy(job, @server)
+    end
+  end
+
+  describe "#do_dispatch" do
+    let(:ems_id) { 1 }
+    let(:job) { Job.create_job("VmScan", :name => "Hello, World") }
+
+    before(:each) do
+      dispatcher.instance_variable_set(:@zone, @server.my_zone)
+      dispatcher.instance_variable_set(:@active_container_scans_by_zone_and_ems, @server.my_zone => {ems_id => 0})
+    end
+
+    it "updates 'dispatch_status' attribute of job record to 'active'" do
+      expect(job.dispatch_status).not_to eq "active"
+      dispatcher.do_dispatch(job, ems_id)
+      expect(job.dispatch_status).to eq "active"
+    end
+
+    it "updates ':started_on' attribute of job record" do
+      expect(job.started_on).to be nil
+      Timecop.freeze do
+        timestamp = Time.now.utc
+        dispatcher.do_dispatch(job, ems_id)
+        expect(job.started_on).to eq timestamp
+      end
+    end
+
+    it "increases counter of active container scans by zone and ems by 1" do
+      counter_by_zone_ems = dispatcher.instance_variable_get(:@active_container_scans_by_zone_and_ems)
```

Comment:
Maybe we should expose the ivars through `attr_accessor`s instead of doing instance_variable_get and set here and above/below